src/library/trackset/baseplaylistfeature.cpp

```diff
@@ -179,17 +179,9 @@ void BasePlaylistFeature::activateChild(const QModelIndex& index) {
     if (playlistId == kInvalidPlaylistId) {
         return;
     }
-
     m_pPlaylistTableModel->setTableModel(playlistId);
     emit showTrackModel(m_pPlaylistTableModel);
     emit enableCoverArtDisplay(true);
-    // Update selection
```

Comment:
`emit featureSelect()` is invoked in 2.3 and everything works as expected. What was the reason for not emitting this signal?

Reply:
> `emit featureSelect()` is invoked in 2.3 and everything works as expected.

you're confusing it with activatePlaylist()?
https://github.com/mixxxdj/mixxx/blob/05016f8c223b836977e409862b3e6d6feceff41d/src/library/baseplaylistfeature.cpp#L155-L163
Also `featureSelect()` is redundant here, and it's currently only updating the selection NOT the currentIndex(), unlike when the selection is changed by a (real or emulated) key press inside `WLibrarySidebar::keyPressEvent`.
> it's currently only updating the selection NOT the currentIndex()

I fixed that with the last commit.